### PR TITLE
fix: Connection issue with wallets (PublicKey)

### PR DESCRIPTION
**Pull request by HarisSQA**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

# Type of change.

- [x] FIX - a PR of this type patches a bug.
- [ ] FEATURE - a PR of this type introduces a new feature.
- [ ] BUILD - a PR of this type introduces build changes.
- [ ] CI - a PR of this type introduces CI changes.
- [ ] DOCS - a PR of this type introduces DOCS improvement.
- [ ] STYLE - a PR of this type introduces style changes.
- [ ] REFACTOR - a PR of this type introduces refactoring.
- [ ] PERFORMANCE - a PR of this type introduces performance changes.
- [ ] TEST - a PR of this type adds more tests.
- [ ] CHORE - a PR introduces other changes than the specified above.

### CODE REVIEW

File: packages/my-near-wallet/src/lib/my-near-wallet.ts [hunk 1]
```diff
@@ -16,6 +16,7 @@ import type {
 } from "@near-wallet-selector/core";
 import { createAction } from "@near-wallet-selector/wallet-utils";
 import icon from "./icon";
+import edgeSSRLoader from "next/dist/build/webpack/loaders/next-edge-ssr-loader";
```
Comment (kujtimprenkuSQA) [inline packages/my-near-wallet/src/lib/my-near-wallet.ts:19]: Why is this import statement needed here?

File: packages/my-near-wallet/src/lib/my-near-wallet.ts [hunk 2]
```diff
@@ -75,26 +76,36 @@ const MyNearWallet: WalletBehaviourFactory<
...
-            account.accountId,
-            options.network.networkId
-          )
-        ).toString(),
-      },
-    ];
+    const publicKeyTemp = await account.connection.signer.getPublicKey(
+      account.accountId,
+      options.network.networkId
+    );
```
Comment (kujtimprenkuSQA) [inline packages/my-near-wallet/src/lib/my-near-wallet.ts:89]: It looks like `signIn` without a `contractId` is still not working for **MyNearWallet** the error is not showing but you can not sign in either:

I would suggest that we change the `getAccounts` function into something like this:

```ts
const getAccounts = async (): Promise<Array<Account>> => {
    const accountId = _state.wallet.getAccountId();
    const account = _state.wallet.account();
    if (!accountId || !account) {
      return [];
    }
    try {
      const publicKey = await account.connection.signer.getPublicKey(
        account.accountId,
        options.network.networkId      );
      return [
        {
          accountId,
          publicKey: publicKey.toString(),
        },
      ];
    } catch (e) {
      return [
        {
          accountId,
          publicKey: undefined,
        },
      ];
    }
  };
```

In the past, the `getAccounts` method was not `async` so the best we can do for now is check and catch any error while trying to get the `publicKey` when there are no errors we return the values for `accountId` and `publicKey` if there are errors for `publicKey` we return `undefined`.

**Review by kujtimprenkuSQA — CHANGES_REQUESTED** [review]
Please check also other wallets since we added the `publicKey` in the Account interface the `getAccounts` has changed from a synchronous method to an asynchronous method.

For example Sender throws an error in this case if the wallet is `Locked`

File: packages/my-near-wallet/src/lib/my-near-wallet.ts (hunk 1 above)
Comment (HarisSQA) [reply packages/my-near-wallet/src/lib/my-near-wallet.ts:19]: I didn't realize that. Thank you!

File: packages/my-near-wallet/src/lib/my-near-wallet.ts (hunk 2 above)
Comment (HarisSQA) [reply packages/my-near-wallet/src/lib/my-near-wallet.ts:89]: Thank you for info and your example of code 👍

**Review by kujtimprenkuSQA — APPROVED** [review]

**Pull request merged**